source/fundamentals/crud/write-operations/delete.txt

```diff
@@ -0,0 +1,150 @@
+
+Sample Data
+~~~~~~~~~~~
+
+To run the example in this guide, load these documents into the
+``ratings`` collection of the ``tea`` database with the following
+snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/delete.go
+   :language: go
+   :dedent:
+   :start-after: begin insertDocs
+   :end-before: end insertDocs
+
+.. tip:: Non-existent Database and Collections
+
+   The driver automatically creates the necessary database and/or collection
+   when you perform a write operation against them if they don't already
+   exist.
+   
+Each document contains a rating for a type of tea, which corresponds to
+the ``type`` and ``rating`` fields.
+
+Delete Operations
+-----------------
+
+Use **delete operations** to remove data from MongoDB. Delete operations
+consist of the ``DeleteOne()`` and ``DeleteMany()`` functions.
+
+The ``DeleteOne()`` function deletes *the first document* that
+matches the filter. The ``DeleteMany()`` function deletes *all*
+documents that match the filter.
+
+.. tip::
+
+   If one document matches your filter, the ``DeleteMany()`` function
+   is equivalent to the ``DeleteOne`` function.
+
+Parameters
+~~~~~~~~~~
+
+Both functions expect you to pass a ``Context`` type and a non-nil query
+filter specifying which documents to match. They both optionally take a
+``DeleteOptions`` type as a third parameter, which represents options
+you can use to configure the delete operation.
+
+The ``DeleteOptions`` type consists of the following functions:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Function
+     - Description
+
+   * - ``SetHint()`` 
+     - | The index to use to scan for documents to delete. 
+       | Default: ``nil``
+
+   * - ``SetCollation()``
+     - | The type of language collation to use when sorting results.  
+       | Default: ``nil``
+
+Example
+```````
+
+The following snippet specifies the delete operation to use the ``_id``
+as the index:
+
+.. code-block:: go
+
+   deleteOptions := options.Delete().SetHint(bson.D{{"_id", 1}})
+
+Return Value
+~~~~~~~~~~~~
+
+Both functions return a ``DeleteResult`` type. This type contains the
+``DeletedCount`` property, which states number of documents
+deleted. If there are no matches to your filter, no document gets
+deleted and ``DeletedCount`` is ``0``.
+
+Example
+```````
+
+The following example passes a context and a filter to the
+``DeleteMany()`` function, which matches and deletes documents where the
+``rating`` is greater than ``8``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/delete.go
+   :language: go
+   :dedent:
+   :start-after: begin deleteMany
+   :end-before: end deleteMany
+
+After running the preceding example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   Number of documents deleted: 2
+
+.. tip::
+
+   If the preceding example used the ``DeleteOne()`` function, the
+   driver deletes the first document matched of the two.
```

Comment:
If the preceding example used the ``DeleteOne()`` function instead of ``DeleteMany()``, the
   driver would delete the first of the two matched documents.